### PR TITLE
feat: signinpage, signinservice

### PR DESCRIPTION
**Pull request by tushartiwari7**
PR for #12 

- Added Signin Page
- A new user will be added to DB on signin.

localhost_3000_signup

### CODE REVIEW

File: src/pages/Signup/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import { BsArrowRight } from "react-icons/bs";
+import { postNewUser } from "../../services";
+import { useUser } from "../../context";
+import { Link, useNavigate } from "react-router-dom";
+
+export const Signup = () => {
+  const { setUser } = useUser();
+  const navigator = useNavigate();
+
+  const signupHandler = async (e) => {
```
Comment (Manju-sree) [inline src/pages/Signup/Signup.jsx:11]: good to see  you used iffi function

**Review by Manju-sree — APPROVED** [review]
I like the the way you have written clean code . Is it still in a WIP, bcoz in preview i can't see it is working , i feel it's static still and i also suggest do commits more in number, as i can see all changes are done in single commit.

File: src/pages/Signup/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    e.preventDefault();
+    console.log(e.target.elements);
```
Comment (tusharkandpal) [inline src/pages/Signup/Signup.jsx]: please remove this `console.log`

File: src/pages/Signup/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        email.value,
+        password.value
+      );
+
+      localStorage.setItem("token", JSON.stringify(data.encodedToken));
+      const createdUser = data.createdUser;
+      delete createdUser.password;
+      setUser({ ...createdUser, isLoggedIn: true });
+      navigator("/products", { replace: true });
+      console.log({ data, status });
```
Comment (tusharkandpal) [inline src/pages/Signup/Signup.jsx]: this too

**Review by tusharkandpal — APPROVED** [review]
i liked the way you organized all the files. clean & easy to understand. 👍🏻

**Review by Bhakti199 — APPROVED** [review]
- This code looks fine to me. 
- Proper hierarchy is maintained.
- proper variable names.
- Remove extra `console.log`
- Overall code is nicely written and structured properly.
This PR is good to merge.

File: src/pages/Signup/Signup.jsx (hunk 1 above)
Comment (tushartiwari7) [reply src/pages/Signup/Signup.jsx:11]: Thanks, don't know why but it's not working on deployed link but working on localhost. Will resolve it soon.

File: src/pages/Signup/Signup.jsx (hunk 3 above)
Comment (tushartiwari7) [reply src/pages/Signup/Signup.jsx]: done

File: src/pages/Signup/Signup.jsx (hunk 2 above)
Comment (tushartiwari7) [reply src/pages/Signup/Signup.jsx]: done

**Pull request merged**